src/isomorphic/classic/element/ReactElement.js

```diff
@@ -284,14 +299,20 @@ ReactElement.cloneElement = function(element, config, children) {
         'Properties defined in its prototype chain will be ignored.'
       );
     }
-    if (config.ref !== undefined) {
+
+    var configRef = getRef(config);
+    var configKey = getKey(config);
+
+    if (configRef != null) {
       // Silently steal the ref from the parent.
```

Comment:
Any specific reason why you're changing comparisons to be looser here? They used to be true for null but are now false.